### DISCUSSION TITLE
Uploading 6.5" and 5.8" iPhone screenshots causes error

### DISCUSSION THREAD

**Discussion by chadhobson** [fastlane/fastlane#18046] [Q&A]
My understanding from the deliver docs is that I have a couple options when uploading screenshots...

1. Upload one set of 6.5" images (1284 x 2778) which Apple will then scale for the 5.8" images.
2. Upload two sets of images, one for 6.5" (1284 x 2778) and one for 5.8" (1170 x 2532).

The first option works flawlessly with deliver but my app actually presents slightly different on the two phones and since I'm automating all of the screenshots anyway, I figured why not go with option 2?

I verified I can upload the 5.8" images manually via iTunes Connect, but the moment I have **both** sets of images being sent via deliver, I get this error.

> [16:26:54]: [33m🚫 Invalid screenshots were detected! Here are the reasons:[0m
[16:26:54]: [31m🚫 Error: ./fastlane/screenshots/ios/en-US/0_LOGIN_1170w_2532h.png - Invalid screen size (Actual size is 1170x2532. See the specifications to fix https://help.apple.com/app-store-connect/#/devd274dd925)[0m

Again, the 5.8" images are 1170x2532 which is correct (https://help.apple.com/app-store-connect/#/devd274dd925) and I can manually upload it through ITC so...

Am I missing something obvious? Here's my deliver command...

`deliver({
            app_identifier: "com.company.MyAwesomeApp",
            screenshots_path: "./fastlane/screenshots/ios",
            skip_binary_upload: true,
            skip_metadata:true,
            skip_app_version_update: true,
            overwrite_screenshots: true,
            submit_for_review: false
        })`

Thanks!

**Comment by gleue** [comment]
Same here. But it used to work in earlier versions.

I patched `/Library/Ruby/Gems/2.6.0/gems/fastlane-2.172.0/deliver/lib/deliver/app_screenshot.rb` to include 1170x2532 sizes in array elements `ScreenSize::IOS_58_MESSAGES` and `ScreenSize::IOS_58` in methods `self.device_messages` and `self.devices` respectively.

That worked for me...

**Reply by chadhobson** [reply]
@gleue I'm not that familiar with Ruby. In javascript we have a package called "patch-package" to handle this stuff. Are you using an equivalent or doing it some other way?

**Reply by gleue** [reply]
I simply patched (aka edited) the file by hand ;-)
